kukkuu/oidc.py

```diff
@@ -0,0 +1,148 @@
+import logging
+from typing import Optional
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from helusers.authz import UserAuthorization
+from helusers.jwt import JWT, ValidationError
+from helusers.oidc import AuthenticationError, RequestJWTAuthentication
+from helusers.settings import api_token_auth_settings
+from helusers.user_utils import get_or_create_user
+from jose import jwt as jose_jwt
+
+from kukkuu.tests.utils.jwt_utils import is_valid_256_bit_key
+
+logger = logging.getLogger(__name__)
+
+
+class ApiTokenAuthSettings:
+    def __init__(self, **entries):
+        self.__dict__.update(entries)
+
+
+class BrowserTestAwareJWTAuthentication(RequestJWTAuthentication):
+    def __init__(self):
+        super().__init__()
+        self._api_token_auth_settings = ApiTokenAuthSettings(
+            **{
+                **api_token_auth_settings._settings,
+                **settings.OIDC_BROWSER_TEST_API_TOKEN_AUTH,
+            }
+        )
```

Comment:
A small thing but how about splitting this to make it more beautiful
(Beauty is not purely objective but IMHO at least)?
```suggestion
        combined_settings = {
            **api_token_auth_settings._settings,
            **settings.OIDC_BROWSER_TEST_API_TOKEN_AUTH,
        }
        self._api_token_auth_settings = ApiTokenAuthSettings(**combined_settings)
```

Reply:
So I'd say these are in favor of this change:
- Beautiful is better than ugly.
- Flat is better than nested.
- Readability counts.

Although the beautiful & readability are not as clear cut as the flat/nested factor. But this code is at least more flat as the previous one had nested `**` operator use.

Reply:
Agreed